src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -310,22 +310,23 @@ public sealed class SecurityIdentifier : IdentityReference, IComparable<Security
         //       the typecast
         //
 
-        private void CreateFromParts(IdentifierAuthority identifierAuthority, int[] subAuthorities)
+        private void CreateFromParts(IdentifierAuthority identifierAuthority, ReadOnlySpan<int> subAuthorities)
         {
-            if (subAuthorities == null)
+            if (subAuthorities.IsEmpty)
             {
                 throw new ArgumentNullException(nameof(subAuthorities));
             }
 
+            int subAuthoritiesLength = subAuthorities.Length;
```

Comment:
Why did you create a local for this?  Seems unnecessary, and on top of that it can actually defeat JIT optimizations.

Reply:
I thought I was being helpful. If they JIT wants to do the work of identifying multiple uses of the value I'll let it. Changed.